### ISSUE TITLE
In Yarn Workspace projects, including the sentry-expo/upload-sourcemaps postPublish hook causes a "project not found" error in remote EAS build process

### ISSUE THREAD

**Issue by connor-odoherty** [expo/sentry-expo#233]
### Summary

If the Sentry "postPublish" configuration is included in app.config.json, running `eas build` fails during the "Run fastlane" step with the error:

```
› Executing Moment » Bundle React Native code and images
    the transform cache was reset.

❌  error: project not found

    the transform cache was reset.
› Executing Moment » [CP] Copy Pods Resources
› Executing Moment » Upload Debug Symbols to Sentry

❌  error: project not found

▸ ** ARCHIVE FAILED **
▸ The following build commands failed:
▸ 	PhaseScriptExecution Upload\ Debug\ Symbols\ to\ Sentry /Users/expo/Library/Developer/Xcode/DerivedData/Moment-azygcsoclksdzbhguedqttdwrjko/Build/Intermediates.noindex/ArchiveIntermediates/Moment/IntermediateBuildFilesPath/Moment.build/Release-iphoneos/Moment.build/Script-2A792CDEC30C4006A3A29DA6.sh (in target 'Moment' from project 'Moment')
▸ (1 failure)
** ARCHIVE FAILED **
```

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

iOS

### SDK Version (managed workflow only)

44

### Environment

```
  Expo CLI 5.0.3 environment info:
    System:
      OS: macOS 11.5.2
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 16.13.1 - ~/.nvm/versions/node/v16.13.1/bin/node
      Yarn: 1.22.4 - ~/.yarn/bin/yarn
      npm: 8.1.2 - ~/.nvm/versions/node/v16.13.1/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: iOS 15.0, DriverKit 20.4, macOS 11.3, tvOS 15.0, watchOS 8.0
      Android SDK:
        API Levels: 26, 29, 8
        Build Tools: 22.0.1
        System Images: android-29 | Google Play Intel x86 Atom
    IDEs:
      Android Studio: 4.0 AI-193.6911.18.40.6514223
      Xcode: 13.0/13A233 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^44.0.0 => 44.0.3 
      react-dom: 17.0.1 => 17.0.1 
      react-native: 0.64.3 => 0.64.3 
    npmGlobalPackages:
      eas-cli: 0.45.0
    Expo Workflow: managed
```

### Reproducible demo or steps to reproduce from a blank project

1. Create a Yarn Workspaces file structure with a Managed EAS Expo app with Sentry Expo
2. Define the required Sentry Expo configurations in app.config.json:
```
    "hooks": {
      "postPublish": [
        {
          "file": "sentry-expo/upload-sourcemaps",
          "config": {
            "organization": xxx,
            "project": xxx
          }
        }
      ]
    },
    "plugins": [
      "sentry-expo"
    ]
```

3. Add the correct symlinks to the expo-yarn-workspaces "symlinks" field of package.json (I've experimented with defining just sentry-expo in addition to the other @sentry packages:

```
...
"expo-yarn-workspaces": {
    "symlinks": [
      ...
      "sentry-expo",
      "@sentry/react-native",
      "@sentry/cli"
    ]
  }
}
...
```
4. Run a remote managed EAS build from the Native app directory: `eas build --platform ios`

5. See the error on the remote console:

```
    › Executing Moment » Bundle React Native code and images
    the transform cache was reset.

❌  error: project not found

    the transform cache was reset.
› Executing Moment » [CP] Copy Pods Resources
› Executing Moment » Upload Debug Symbols to Sentry

❌  error: project not found

▸ ** ARCHIVE FAILED **
▸ The following build commands failed:
▸ 	PhaseScriptExecution Upload\ Debug\ Symbols\ to\ Sentry /Users/expo/Library/Developer/Xcode/DerivedData/Moment-azygcsoclksdzbhguedqttdwrjko/Build/Intermediates.noindex/ArchiveIntermediates/Moment/IntermediateBuildFilesPath/Moment.build/Release-iphoneos/Moment.build/Script-2A792CDEC30C4006A3A29DA6.sh (in target 'Moment' from project 'Moment')
▸ (1 failure)
** ARCHIVE FAILED **
```

Other notes:
* Removing the post-publish hook from app.config.json results in the build working perfectly without error
* I have attempted both Yarn Workspace configurations suggested by https://github.com/expo/expo/tree/master/packages/expo-yarn-workspaces and https://github.com/byCedric/eas-monorepo-example/tree/main/apps/with-sentry . They differ in many ways (the Metro configuration, app entry point, etc.), so I am not sure which is best practice at this point. Both approaches resulted in the above error, though

**Comment by stuckj** [comment]
This is happening for me as well (on SDK 43 though). EAS builds and managed workflow. We recently broke our app out of a mono repo in the hopes of fixing this (and reducing other complexity). The problem still remains when not in a mono repo as well though.

**Comment by stuckj** [comment]
> This is happening for me as well (on SDK 43 though). EAS builds and managed workflow. We recently broke our app out of a mono repo in the hopes of fixing this (and reducing other complexity). The problem still remains when not in a mono repo as well though.

Oh man. It turned out for us the problem was very simple. Someone had renamed the project name in sentry's UI so it no longer matched the project name in the `postPublish` configuration. Be sure to check that your project name matches if you're hitting this. :-P

**Comment by loureirorg** [comment]
In my case was also a name mismatch. There are three places the name should be the same: 
1. `expo.name` in the `app.json` file;
2. `postPublish` configuration, also in the `app.json` file;
3. Project name in Sentry's UI;

The name in these three parts MUST be all lowercase and just `[a-z]`. No spaces, no dashes, no uppercases, no special characters.

Good examples: `myproject`, `myamazingprojectnumbertwo`.
Bad examples: `My Project`, `MyProject`, `my-project`, `my_project`

In my case, I went to Sentry's UI and set the project name to `my-project`. Sentry UI doesn't allow for uppercases, so I used dashes to separate words. As the project name in the `postPublish` must be the same, I set it to `my-project`. Still, EAS building was failing with the ` error: project not found ** ARCHIVE FAILED **` error. 

I noticed in the error logs that EAS was using the string `MyProject` in the script, which I haven't set anywhere. And this mismatch between `MyProject` and `my-project` (`postPublish`) was causing the build to fail.

I found that EAS generates the project name from `expo.name`, which I had set to `My Project`. It removes spaces, dashes, and special characters.

Why EAS doesn't use `expo.slug` I have no idea.

So, to conform with both EAS and Sentry's UI name rules, you have to only use lowercase ASCII, with no word separation.

---

In short: 
1. `expo.name`, `postPublish`, and Sentry's UI MUST have the same value;
2. Use lowercase for your project name, with no word separation. Ex. `myamazingprojectnumbertwo`;

**Comment by loureirorg** [comment]
In my case, the `organization` field on `postPublish` didn't match Sentry's UI. 

I changed the organization and project's name on Sentry and forgot to update on `app.json`.
